File: data-ingestion/src/data_ingestion/app/api/api_v1/models/weather.py
from pydantic import BaseModel


class CurrentWeatherProps(BaseModel):
    city: str
    country_code: str


class CurrentWeatherDTO(BaseModel):
    city: str
    sunrise: int 
    sunset: int
    weather_condition: str 
    weather_icon_url: str
    temp: float
    feels_like: float
    temp_min: float
    temp_max: float
    pressure: int
    humidity: int
    wind_speed: float 
    wind_deg: float
    

File: ion-frontend/src/data/schema/weather.ts
export interface CurrentWeatherSchema {
    city: string;
    sunrise: number;
    sunset: number;
    weather_condition: string;
    weather_icon_url: string;
    temp: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    humidity: number;
    wind_speed: number;
    wind_deg: number;
}
